libr/debug/p/native/linux/linux_debug.c

```diff
@@ -584,37 +624,17 @@ static RList *get_pid_thread_list(RDebug *dbg, int main_pid) {
 	return list;
 }
 
-static void linux_attach_all(RDebug *dbg) {
-	linux_attach_single_pid (dbg, dbg->main_pid);
-
-	RList *list = dbg->threads;
-	if (list) {
-		RDebugPid *th;
-		RListIter *it;
-		r_list_foreach (list, it, th) {
-			if (th->pid && th->pid != dbg->main_pid) {
-				if (!linux_attach_single_pid (dbg, th->pid)) {
-					eprintf ("PID %d\n", th->pid);
-				}
-			}
-		}
-	}
-}
-
 int linux_attach(RDebug *dbg, int pid) {
 	// First time we run: We try to attach to all "possible" threads and to the main pid
 	if (!dbg->threads) {
 		dbg->threads = get_pid_thread_list (dbg, pid);
-		linux_attach_all (dbg);
 	} else {
 		// This means we did a first run, so we probably attached to all possible threads already.
 		// So check if the requested thread is being traced already. If yes: skip
 		if (dbg->threads && !r_list_find (dbg->threads, &pid, &match_pid)) {
-			goto out;
+			linux_attach_single_pid (dbg, pid);
```

Comment:
You are right. I was confused by the function of `match_pid` at first.

Reply:
The statement actually runs when it did not find an existing thread because `!r_list_find (dbg->threads, &pid, &match_pid)` will be true when `r_list_find` returns `NULL` and there is no existing thread with the same pid.

Reply:
Before this, `match_pid` is implemented differently and correcting it reverses the condition. I will update the comment instead.